ams/server/src/main/java/com/netease/arctic/server/dashboard/ServerTableDescriptor.java

```diff
@@ -225,29 +227,71 @@ public List<DDLInfo> getTableOperations(ServerTableIdentifier tableIdentifier) {
     return result;
   }
 
-  public List<OptimizedRecord> getOptimizeInfo(String catalog, String db, String table) {
-    List<TableOptimizingProcess> tableOptimizingProcesses = getAs(
+  public PageResult<TableOptimizingProcess> getOptimizingProcesses(String catalog, String db, String table, int offset,
+                                                                   int limit) {
+    List<TableOptimizingProcess> tableOptimizingProcesses = Collections.emptyList();
+    if (limit > 0) {
+      // do this because of derby not support LIMIT
+      List<Long> allProcessIds = getAs(
+          OptimizingMapper.class,
+          mapper -> mapper.selectOptimizingProcessIds(catalog, db, table));
+      List<Long> processIds = allProcessIds.stream()
+          .sorted(Comparator.reverseOrder())
+          .skip(offset)
+          .limit(limit)
+          .collect(Collectors.toList());
+      if (!processIds.isEmpty()) {
+        tableOptimizingProcesses = getAs(
+            OptimizingMapper.class,
+            mapper -> mapper.selectOptimizingProcesses(catalog, db, table, processIds.get(processIds.size() - 1),
+                processIds.get(0)));
+        List<OptimizingTaskStat> optimizingTaskStats = getAs(OptimizingMapper.class,
+            mapper -> mapper.selectOptimizeTaskStats(processIds));
+        initOptimizingProcesses(tableOptimizingProcesses, optimizingTaskStats);
```

Comment:
This processing method is very inefficient in MySQL. Is there any other way we can do it?

Reply:
For MySQL, the `SELECT ... IN (items)` query using the primary key usually does not have performance issues, especially when the number of `items` is small. ( In this case, the number of `items` is page size, usually less than 100).